File: Java/src/main/java/com/gildedrose/GildedRose.java
package com.gildedrose;

import com.gildedrose.items.Item;

class GildedRose {
    Item[] items;

    public GildedRose(Item[] items) {
        this.items = items;
    }

    public void updateQuality() {
        for (var item : items) {
            var specializedItem = new ItemFactory(item).getSpecificItemFromName(item);
            specializedItem.updateState();
        }
    }
}

File: Python/src/gilded_rose.py
from src.item_factory import get_specific_item_from_name


class GildedRose(object):

    def __init__(self, items):
        self.items = items

    def update_quality(self):
        for item in self.items:
            specialized_item = get_specific_item_from_name(item)
            specialized_item.update_state()
